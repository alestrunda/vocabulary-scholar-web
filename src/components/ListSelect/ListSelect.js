import React from 'react';
import PropTypes from 'prop-types';
import NavigationCheck from 'material-ui/svg-icons/navigation/check';
import classNames from 'classnames';

const ListSelect = props => {
  const handleClick = () => {
    props.onClick(props.list.id, props.isSelected);
  };

  return (
    <div
      className={classNames(
        'list-select',
        {
          active: props.isSelected,
        },
        props.className
      )}
      onClick={handleClick}
      style={props.style}
    >
      <div className="list-select__icon">
        {props.isSelected && <NavigationCheck color="#fff" />}
      </div>
      <h3 className="text-wrap">{props.list.name}</h3>
      <p>Words: {props.list.wordIDs.length}</p>
    </div>
  );
};

ListSelect.propTypes = {
  className: PropTypes.string,
  list: PropTypes.shape({
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    wordIDs: PropTypes.array.isRequired,
  }).isRequired,
  isSelected: PropTypes.bool,
  onClick: PropTypes.func.isRequired,
  style: PropTypes.object,
};

export default ListSelect;
