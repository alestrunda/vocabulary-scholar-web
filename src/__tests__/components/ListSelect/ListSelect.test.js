import React from 'react';
import { shallow } from 'enzyme';

import ListSelect from '../../../components/ListSelect';

it('renders correctly', () => {
  const list = {
    id: '1',
    name: 'List 1',
    wordIDs: ['vigorous', 'travel', 'car'],
    dateCreated: 1544482800000,
    dateViewed: 1544482800000,
  };
  const onClick = jest.fn();
  const root = shallow(<ListSelect isSelected onClick={onClick} list={list} />);
  expect(root).toMatchSnapshot();
});

it('triggers onClick', () => {
  const list = {
    id: '1',
    name: 'List 1',
    wordIDs: ['vigorous', 'travel', 'car'],
    dateCreated: 1544482800000,
    dateViewed: 1544482800000,
  };
  const isSelected = true;
  const onClick = jest.fn();
  const root = shallow(
    <ListSelect isSelected={isSelected} onClick={onClick} list={list} />
  );
  root.simulate('click');
  expect(onClick).toHaveBeenCalledWith(list.id, isSelected);
});
