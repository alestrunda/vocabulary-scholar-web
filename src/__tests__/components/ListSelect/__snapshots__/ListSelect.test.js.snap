// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ListSelect
    isSelected={true}
    list={
      Object {
        "dateCreated": 1544482800000,
        "dateViewed": 1544482800000,
        "id": "1",
        "name": "List 1",
        "wordIDs": Array [
          "vigorous",
          "travel",
          "car",
        ],
      }
    }
    onClick={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="list-select__icon"
        >
          <NavigationCheck
            color="#fff"
          />
        </div>,
        <h3
          className="text-wrap"
        >
          List 1
        </h3>,
        <p>
          Words: 
          3
        </p>,
      ],
      "className": "list-select active",
      "onClick": [Function],
      "style": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <NavigationCheck
            color="#fff"
          />,
          "className": "list-select__icon",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "color": "#fff",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "List 1",
          "className": "text-wrap",
        },
        "ref": null,
        "rendered": "List 1",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Words: ",
            3,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Words: ",
          3,
        ],
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="list-select__icon"
          >
            <NavigationCheck
              color="#fff"
            />
          </div>,
          <h3
            className="text-wrap"
          >
            List 1
          </h3>,
          <p>
            Words: 
            3
          </p>,
        ],
        "className": "list-select active",
        "onClick": [Function],
        "style": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NavigationCheck
              color="#fff"
            />,
            "className": "list-select__icon",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "color": "#fff",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "List 1",
            "className": "text-wrap",
          },
          "ref": null,
          "rendered": "List 1",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Words: ",
              3,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Words: ",
            3,
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
